sky/templates/kubernetes-port-forward-proxy-command.sh.j2

```diff
@@ -45,8 +45,26 @@ done
 # To avoid errors when many concurrent requests are sent (see https://github.com/skypilot-org/skypilot/issues/2628),
 # we add a random delay before establishing the socat connection.
 # Empirically, this needs to be at least 1 second. We set this to be random between 1 and 2 seconds.
-sleep $(shuf -i 10-20 -n 1 | awk '{printf "%f", $1/10}')
-
+if which shuf > /dev/null; then
+  sleep $(shuf -i 10-20 -n 1 | awk '{printf "%f", $1/10}')
+elif which gshuf > /dev/null; then
+  sleep $(shuf -i 10-20 -n 1 | awk '{printf "%f", $1/10}')
+else 
+  if [[ "$OSTYPE" == "darwin"* ]]; then
+      # Check if gshuf is installed
+      if [[ $(command -v gshuf) == "" ]]; then
+          echo "gshuf is not installed. Please install gshuf first."
+          exit 1
+      fi
+  else
+      # Check if shuf is installed
+      if [[ $(command -v shuf) == "" ]]; then
+          echo "shuf is not installed. Please install shuf first."
+          exit 1
+      fi
+      echo "shuf is installed."
+  fi
+fi
```

Comment:
For consistency, we should move these checks to after L14, like we have for `socat` and `nc`. We should also probably detect the right binary (`gshuf` vs `shuf`) during that check and use that later on here at L48.